### PR TITLE
Feature/stop scan oxo api

### PR DESCRIPTION
**Pull request by oussamaessaji**

### CODE REVIEW

File: src/ostorlab/serve_app/oxo.py [hunk 1]
```diff
@@ -177,8 +178,46 @@ def mutate(
             return DeleteScanMutation(result=True)
 
 
+class StopScanMutation(graphene.Mutation):
+    """stop scan mutation."""
+
+    class Arguments:
+        scan_id = graphene.String(required=True)
```
Comment (elyousfi5) [inline src/ostorlab/serve_app/oxo.py:185]:
```suggestion
        scan_id = graphene.Int(required=True)
```

File: src/ostorlab/serve_app/oxo.py [hunk 2]
```diff
@@ -177,8 +178,46 @@ def mutate(
...
+        Raises:
+            graphql.GraphQLError in case the scan does not exist or the scan id is invalid.
+
+        Returns:
+            The stopped scan.
+
+        """
+        with models.Database() as session:
+            try:
+                int_scan_id = int(scan_id)
```
Comment (elyousfi5) [inline src/ostorlab/serve_app/oxo.py:205]: not needed !

File: tests/serve_app/oxo_test.py [hunk 3]
```diff
@@ -459,3 +459,59 @@ def testDeleteScanMutation_whenScanDoesNotExist_returnErrorMessage(
...
+  stopScan(scanId: $scanId){
+    scan{
+      id
+    }
+  }
+}
+        """
+        response = client.post(
+            "/graphql", json={"query": query, "variables": {"scanId": str(scan.id)}}
+        )
```
Comment (elyousfi5) [inline tests/serve_app/oxo_test.py:484]:
```suggestion

        response = client.post(
            "/graphql", json={"query": query, "variables": {"scanId": str(scan.id)}}
        )
```

File: tests/serve_app/oxo_test.py [hunk 4]
```diff
@@ -459,3 +459,59 @@ def testDeleteScanMutation_whenScanDoesNotExist_returnErrorMessage(
 
     assert response.status_code == 200, response.get_json()
     assert response.get_json()["errors"][0]["message"] == "Scan not found."
+
+
+def testStopScanMutation_always_shouldStopScan(
```
Comment (benyissa) [inline tests/serve_app/oxo_test.py:464]:
```suggestion
def testStopScanMutation_whenScanIsRunning_shouldStopScan(
```

The second test has a specific condition. so always is not useful in this case.

Comment (oussamaessaji) [reply tests/serve_app/oxo_test.py:464]: make sense

File: src/ostorlab/serve_app/oxo.py (hunk 1 above)
Comment (oussamaessaji) [reply src/ostorlab/serve_app/oxo.py:185]: The stop method requires a string.

File: src/ostorlab/serve_app/oxo.py (hunk 2 above)
Comment (oussamaessaji) [reply src/ostorlab/serve_app/oxo.py:205]: The stop method requires a string.

**Review by benyissa — CHANGES_REQUESTED** [review]
This branch has conflicts that must be resolved

File: src/ostorlab/serve_app/oxo.py (hunk 1 above)
Comment (3asm) [reply src/ostorlab/serve_app/oxo.py:273]: The API must match. You should use Int.

File: src/ostorlab/serve_app/oxo.py [hunk 5]
```diff
@@ -265,8 +266,46 @@ def mutate(
...
+            except ValueError:
+                raise graphql.GraphQLError("%s not a valid scan id.", int_scan_id)
+
+            scan = session.query(models.Scan).get(int_scan_id)
+            if scan is None:
+                raise graphql.GraphQLError("Scan not found.")
+            local_runtime.LocalRuntime().stop(scan_id=scan_id)
```
Comment (3asm) [inline src/ostorlab/serve_app/oxo.py:300]: Do str.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/serve_app/oxo.py [hunk 6]
```diff
@@ -265,8 +266,46 @@ def mutate(
             return DeleteScanMutation(result=True)
 
 
+class StopScanMutation(graphene.Mutation):
+    """stop scan mutation."""
```
Comment (deadly-panda) [inline src/ostorlab/serve_app/oxo.py:270]:
```suggestion
    """Stop scan mutation."""
```

File: src/ostorlab/serve_app/oxo.py [hunk 7]
```diff
@@ -265,8 +266,46 @@ def mutate(
...
+            graphql.GraphQLError in case the scan does not exist or the scan id is invalid.
+
+        Returns:
+            The stopped scan.
+
+        """
+        with models.Database() as session:
+            try:
+                int_scan_id = int(scan_id)
+            except ValueError:
```
Comment (deadly-panda) [inline src/ostorlab/serve_app/oxo.py:294]: This is already handled in the stop method of the local runtime..

File: tests/conftest.py [hunk 8]
```diff
@@ -610,6 +610,20 @@ def web_scan(clean_db: None) -> None:
...
+    """Create a dummy web scan."""
+    with models.Database() as session:
+        scan = models.Scan(
+            title="Web Scan",
+            asset="Web",
+            progress=models.ScanProgress.IN_PROGRESS,
+            created_time=datetime.datetime.now(),
+        )
+        session.add(scan)
+        session.commit()
```
Comment (deadly-panda) [inline tests/conftest.py:624]: `Scan` model already exposes a create method which handles this db session logic.

**Review by 3asm — DISMISSED** [review]

**Comment by 3asm** [comment]
Please resolve conflict and merge.

File: tests/conftest.py (hunk 8 above)
Comment (benyissa) [reply tests/conftest.py:626]: +1

File: src/ostorlab/serve_app/oxo.py (hunk 1 above)
Comment (amine3) [reply src/ostorlab/serve_app/oxo.py:273]: To switch to int

File: src/ostorlab/serve_app/oxo.py (hunk 2 above)
Comment (amine3) [reply src/ostorlab/serve_app/oxo.py:293]: To remove

**Review by amine3 — CHANGES_REQUESTED** [review]

**Review by amine3 — APPROVED** [review]

**Review by mohsinenar — APPROVED** [review]

**Pull request merged**